base/essentials.jl

```diff
@@ -312,10 +312,34 @@ struct Colon
 end
 const (:) = Colon()
 
-# For passing constants through type inference
-struct Val{T}
+"""
+    Val(c)
+
+Constructs a `Val{c}()`, which contains no run-time data. Types like this can be used to
```

Comment:
"Construct" instead of "Constructs" to comply with the usual phrasing of docstrings?